computation_dag/topology_builder.py

```diff
@@ -15,8 +16,14 @@ def _validate_path(self):
 
     def _try_load(self, f, **kwargs):
         try:
-            df = f(path=self.path, **kwargs)
-            df.first()
+            if isinstance(self.path, str):
+                df = f(path=self.path, **kwargs)
+                df.first()
+            else:
```

Comment:
You assume that if it's not a `str` then it must be an iterable (list of tuple).

wouldn't it be more cautious to explicitly check the types and see if it's a list or something else? 
(and in case it's something you don't know how to handle, raise an exception)